hyperspy/_signals/hologram_image.py

```diff
@@ -0,0 +1,274 @@
+
+        # Find output shape:
+        if output_shape is None:
+            output_shape = self.axes_manager.signal_shape
+
+        # ???
+        _logger.info('Sideband pos in pixels: {}'.format(sb_position))
+        _logger.info('Sideband aperture radius in pixels: {}'.format(sb_size))
+        _logger.info('Sideband aperture smoothness in pixels: {}'.format(sb_smooth))
+
+        # Shows the selected sideband and the position of the sideband
+        # if plotting:
+        #     fft_holo = fft2(self.data) / np.prod(self.data.shape)
+        #     fig, axs = plt.subplots(1, 1, figsize=(4, 4))
+        #     axs.imshow(np.abs(fft_holo), clim=(0, 2.2))
+        #     axs.scatter(sb_position[1], sb_position[0], s=20, color='red', marker='x')
+        #     axs.set_xlim(sb_position[1]-sb_size, sb_position[1]+sb_size)
+        #     axs.set_ylim(sb_position[0]-sb_size, sb_position[0]+sb_size)
+        #     plt.show()
+
+        # object electron wave:
+        if self.axes_manager.navigation_size:
+            wave_object = np.zeros((self.axes_manager.navigation_size, ) + output_shape, dtype='complex')
+            for i in range(self.axes_manager.navigation_size):
+                wave_object[i] = reconstruct(self.inav[i].data, holo_sampling=self.sampling,
+                                             sb_size=sb_size[i], sb_position=sb_position[i], sb_smoothness=sb_smooth[i],
+                                             output_shape=output_shape, plotting=plotting)
+        else:
+            wave_object = reconstruct(self.data, holo_sampling=self.sampling,
+                                      sb_size=sb_size, sb_position=sb_position, sb_smoothness=sb_smooth,
+                                      output_shape=output_shape, plotting=plotting)
+        # Reconstructing reference wave and applying it (division):
+        folded_ref = False
+
+        if reference is None:
+            wave_reference = 1
+        else:
+            # reference electron wave
+            if reference.axes_manager.navigation_size:
+                folded_ref = reference.unfold_navigation_space()
+                wave_reference = np.zeros((reference.axes_manager.navigation_size, ) + output_shape, dtype='complex')
+                for i in range(reference.axes_manager.navigation_size):
+                    wave_reference[i] = reconstruct(reference.inav[i].data, holo_sampling=self.sampling,
+                                                    sb_size=sb_size[i], sb_position=sb_position[i], sb_smoothness=sb_smooth[i],
+                                                    output_shape=output_shape, plotting=plotting)
+
+                if reference.axes_manager.navigation_size != self.axes_manager.navigation_size:  # case when navdim of
+                    # reference is >0 and not equal to that of self:
+
+                    warnings.warn('The navigation size of the reference and the hologram do not match! Reference wave '
```

Comment:
Wouldn't it be better to let the user decide in this case what to do? If the navigation sizes are different chances are that the user is doing something wrong but he may not realise it if he doesn't get an error.